web/src/components/Inputs/Button.tsx

```diff
@@ -6,40 +6,60 @@ import CircularProgress from "../Feedback/CircularProgress";
 interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
   disabled?: boolean;
   loading?: boolean;
+  variant?: "contained" | "text";
   className?: string;
   children: ReactNode;
 }
 
+/**
+ * A button component that can be used to trigger actions.
+ *
+ * @param disabled - Whether the button is disabled or not.
+ * @param loading - Whether the button is loading or not. If true, the button will be disabled.
+ * @param variant - The variant can be either "contained" or "text".
+ * @param className - Additional styles to apply to the button.
+ * @param children - The content of the button.
+ * @param rest - Additional props to pass to the button.
+ */
 export default function Button({
   disabled = false,
   loading = false,
+  variant = "contained",
```

Comment:
Not a requirement to merge, just a thought: you might consider refactoring into `ContainedButton` and `TextButton`. They could just call into this function locally, but it can make the class easier to use.

I would strongly advise going for something like `PrimaryButton` and `SecondaryButton` instead though; I think that is more clear. The other option is expose `className` and let people pass their own hover/bg color styles, but I think having those set in stone is probably best.

Reply:
My idea is to have pre-defined button styles set in stone so that we have as least inconsistency as possible. Having many buttons that are technically the same component but just a different variant of it might pollute the directory more. 

I'm more or less following how MUI does things too: https://mui.com/material-ui/react-button/

Reply:
Fair enough - my main concern is the name. I think "transparent" and "filled" make more sense than "text" and "contained". I don't like MUI's convention here. However, if you feel strongly about it, it's not the end of the world.

For the multiple components, I'm not saying to make a bunch of files, just do:
```ts
export const TransparentButton = (props) => Button({ ...props, variant: "transparent" });
export const FilledButton = (props) => Button({ ...props, variant: "filled" });

const Button = ...
```
This is also just a suggestion though.

Reply:
We could rename it to filled yea but I like having text for the button without a background as it is more descriptive to what it actually is. We are arguing over small stuff at this point lol. Lets just do "filled" and "text" like the Material Design spec specifies

![image](https://user-images.githubusercontent.com/39206666/197663250-67fbbcb2-f387-4215-a627-0811072a07d4.png)

Reply:
Ok sounds good. But does this mean that MUI doesn't follow the material design spec?

Reply:
That's probably a legacy name for MUI. I pulled this screenshot from Material Design v3 which is quite recent